client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -93,6 +93,16 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
     .maximumSize(rpcCacheSize)
     .build().asInstanceOf[Cache[Int, ByteBuffer]]
 
+  val sharedRPCThreadPool: Option[ThreadPoolExecutor] =
+    if (conf.clientRPCThreadPoolShare) {
+      Option.apply(ThreadUtils.newDaemonCachedThreadPool(
+        "LifecycleManager-shared-reserveSlot-pool",
```

Comment:
Since this thread pool do more than `reserve slot`, so we should naming this as `LifecycleManager-shared-pool`?